### ISSUE TITLE
Ajv 'coerceTypes' option should probably not be set for request bodies

### ISSUE THREAD

**Issue by pvince** [exegesis-js/exegesis#73]
I realize the following structure is not great, but it matches a legacy API that we can't change without potentially breaking customer's programs. If I was implementing this today, we would do this differently.

I have a body property that can either be a positive integer (minValue: 1) or it can be a string value representing the name of something.  The integer and the string both uniquely identify something.

```
POST .../object/{objectID}
{
   tagIdentifier: 15
}
```
OR
```
POST .../object/{objectID}
{
   tagIdentifier: "Name of label"
} 
```

I wrote up the definition for tagIdentifier as:
```
Object:
  properties:
    tagIdentifier:
      oneOf:
        - type: string
        - type: number
          format: int64
          minimum: 1
```

Howerver, the following was passing schema validation:
```
POST .../object/{objectID}
{
  tagIdentifier: -1
}
```

Digging into the issue I traced it back to the validator which was coercing the 'tagIdentifier' value to always be a string value, so when it went through validation it was checking this as `"-1"` instead of as the actual `-1` value.

Digging through the code it appears the issue is tied to the initialization of Ajv. https://github.com/exegesis-js/exegesis/blob/41ccdb86df754f310d3e73303caef7617fe7a79f/src/oas3/Schema/validators.ts#L193

This is hardly beautiful, but I was able to test out a fix by adding the following lines right before the ajv initialization:
```
    let coerceTypesOveride = allowTypeCoercion ? 'array' : false;
    if (allowTypeCoercion && parameterLocation.in === 'request') {
        // We do not want to allow type coercion on request bodies in any form.
        coerceTypesOveride = false;
    }
```

Changing the type coercion to NOT coerce arrays fixed a separate issue I was having as well, where an array of enum values was accepting a non-array. It was a lesser issue that I was going to ignore, but this resolves that as well.

Currently I am in crunch trying to get our swagger-node to exegesis-express conversion done, but once that is finished I will see if I can send a pull request to resolve this if you agree with the changes.  Maybe there could be a configuration option to choose when to allow type coercion? I think query and path are pretty much required to do type coercion, but it should be an option for request bodies.

**Comment by jwalton** [comment]
Yeah, you are correct.  We should make coerceTypes an option.  We probably need to leave it turned on for parameters (or do some type coercion ourselves) since parameters are always going to get parsed as strings (which is why I turned it on in the first place) but it shouldn't be on for bodies.

We might even want to make it optional based on the mime type of the body; you would generally want it turned off for JSON (because the type is not ambiguous in JSON), but in multipart/mixed or application/x-www-form-urlencoded, you'd still want type coercion...  OR, maybe instead of making it optional, it should be up to the body parser - so the JSON body parser would not do type coercion, but other body parsers might.
